openedx/MITx-edx-integration-tutor.md

```diff
@@ -0,0 +1,219 @@
+
+### MIT Application Setup
+
+To set up MIT Application:
+
+1. Get the gateway IP for the `EDX_APP`
+   1. Linux users: The gateway IP of the docker-compose networking setup for edx LMS `docker network inspect tutor_dev_default | grep Gateway`.
+   2. OSX users: Use `host.docker.internal`
+
+2. Set up your `.env` file. These settings need particular attention:
+
+   - `OPENEDX_IP`: set to the gateway IP from the first step.
+   - `OPENEDX_API_BASE_URL`: set to `http://<EDX_HOSTNAME>:<PORT>`
+   - `OPENEDX_SERVICE_WORKER_USERNAME`: set to `mit_Application_serviceworker` (unless you changed this)
+   - `OPENEDX_SERVICE_WORKER_API_TOKEN`: set to the token you just generated
+
+### Run the configure_instance command
+
+    docker-compose run --rm web ./manage.py configure_instance linux --gateway <ip> --tutor-dev
+
+  Where `<ip>` is the IP from the first step.(On macOS, specify macos instead of linux. You can also skip --gateway.) You will need to supply passwords for the MITx Application superuser and test learner accounts. **Make a note of the client ID and secret that it will print out at the end.**
+
+
+## Tutor Setup, Part Two
+
+Note that some of these steps require editing the main configuration files for the production instance (which is also used for a local deployment). Most of the settings that need to be adjusted to get integration working are overridden by the default Tutor configuration, so you can't update them by setting `config.yml`. If you're using the development Tutor build, you'll likely need to edit `development.py` rather than `production.py` as necessary.
+
+These steps will also disable the AuthN SSO MFE, so from here on you'll get normal edX authentication screens (if you're not being bounced to MITx Application).
+
+
+1. Get the gateway IP of the `mitxApplication_default` Docker network. Example:
+
+       docker network inspect mitxpro_default | grep Gateway
+
+2. Log into to edX using your superuser account, and make sure your session stays open. Sessions are pretty long-lived so this just means not closing the browser that you started the session in. (Part of this process will involve mostly breaking authentication so it's important that you are able to get into the admin.)
+3. Stop Tutor: `tutor dev stop`
+4. Add extra requirements required for OAuth Configuration:
+
+       tutor config save --append OPENEDX_EXTRA_PIP_REQUIREMENTS=social-auth-mitxpro
+       tutor config save --append OPENEDX_EXTRA_PIP_REQUIREMENTS=openedx-companion-auth
+
+   **NOTE**: Reference for [Installing extra xblocks and requirements](https://docs.tutor.edly.io/configuration.html#installing-extra-xblocks-and-requirements)
+
+5. **SKIP NEXT 3 STEPS ONLY if** you have cloned `edx-platform` and mounted it while setting up the tutor. You can create `private.py` file at `edx-platform/lms/envs/{here}` and add the following configurations to allow additional OAuth providers
```

Comment:
This wording is confusing - I had to reread this a few times to understand that it meant "don't do this step for your regular `tutor dev` build". 

Suggestion: `**If you've cloned and mounted `edx-platform` while setting up Tutor**, do this step and skip steps 6, 7, and 8.` 

Or: throw this under Other Notes with its own heading. (And whatever other detail is necessary for running Tutor with cloned openedx - it's a deployment option for Tutor, but it really isn't covered by this doc, so having this step at this part of the process when it's not been brought up before makes it sort of comes out of left field.)

Reply:
Thanks for thoughtful comment @jkachel I really appreciate it.
I will update it as you suggested: `**If you've cloned and mounted edx-platform while setting up Tutor**, do this step and skip steps 6, 7, and 8`